tests/e2e/data/services/test_bq_datasets.py

```diff
@@ -15,9 +16,81 @@
 60,10,60010950600,9506.0,POINT (-170.701028 -14.252446)
 """
 
+CSV_SAMPLE_REDUCED = """id,geom
+1,POINT (-170.5618796 -14.2587411)
+2,POINT (-170.5589852 -14.2859572)
+3,POINT (-170.6310985 -14.2760947)
+4,POINT (-170.6651925 -14.2713653)
+5,POINT (-170.701028 -14.252446)
+"""
 
 class TestBQUserDataset(unittest.TestCase):
 
+    def test_can_upload_from_dataframe(self):
+        sample = StringIO(CSV_SAMPLE_REDUCED)
+        df = pandas.read_csv(sample)
+        unique_table_name = 'cf_test_table_' + str(uuid.uuid4()).replace('-', '_')
+        BQUserDataset.name(unique_table_name).upload(df)
+
+    def test_can_upload_from_file_object(self):
+        unique_table_name = 'cf_test_table_' + str(uuid.uuid4()).replace('-', '_')
+        file_object = StringIO(CSV_SAMPLE_REDUCED)
+        BQUserDataset.name(unique_table_name).upload_file_object(file_object)
+
+    def test_can_import_a_dataset(self):
+        unique_table_name = 'cf_test_table_' + str(uuid.uuid4()).replace('-', '_')
+        BQUserDataset \
+            .name(unique_table_name) \
+            .column(name='id', type='INT64') \
+            .column('geom', 'GEOMETRY') \
+            .ttl_seconds(30) \
+            .create()
+        file_object = StringIO(CSV_SAMPLE_REDUCED)
+        BQUserDataset.name(unique_table_name).upload_file_object(file_object)
+        job = BQUserDataset.name(unique_table_name).import_dataset()
```

Comment:
Thinking out loud: as a matter of style for usage from CF, I thought of something like
```python
dataset = BQUserDataset.name(unique_table_name)
dataset.upload_file_object(file_object)
job = dataset.import_dataset()
ret = job.result()
```
which can easily be wrapped into a function:
```python
BQUserDataset.name('name').import_file('path_to.csv') # blocks until complete, raise an error if upload or import fails
BQUserDataset.name('name').import_dataframe(df) # same here
```

import is not the best name though (or we need another name)